File: IntentApp/app/src/main/java/com/apssdc/intentapp/MainActivity.java
package com.apssdc.intentapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    /*Intent is a message that is passed b/w components
     * such as activities, contentproviders,broadcastreceivers,services.*/
    /*There are two types of intents 1.Implicit,2.ExplicitIntent*/
    EditText e1,e2,e3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        e1 = findViewById(R.id.edit1);
        e2 = findViewById(R.id.edit2);
        e3 = findViewById(R.id.edit3);
    }

    public void submit(View view) {
        Intent intent = new Intent(this,SecondActivity.class);
        startActivity(intent);


    }

    public void sendUrl(View view) {
        String url = e1.getText().toString();
        Uri webpage = Uri.parse("https://www."+url);
        Intent intent = new Intent(Intent.ACTION_VIEW,webpage);
        startActivity(intent);
    }

    public void passValue(View view) {
        String name = e2.getText().toString();
        Intent intent = new Intent(this,SecondActivity.class);
        intent.putExtra("KEY",name);
        startActivity(intent);
    }

    public void location(View view) {

        String loc = e3.getText().toString();
        Uri address = Uri.parse("geo:0,0?q=" + loc);
        Intent intent = new Intent(Intent.ACTION_VIEW,address);
        startActivity(intent);













    }
}

File: ImplicitIntents_Kri/app/src/main/java/com/example/implicitintents_kri/MainActivity.kt
package com.example.implicitintents_kri

import android.app.Activity
import android.content.Intent
import android.content.Intent.ACTION_VIEW
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.provider.MediaStore
import android.view.View

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun openProfile(view: View) {
        val i = Intent()
        i.setAction(ACTION_VIEW)
        i.setData(Uri.parse("https://svkrishna3724.github.io"))
        startActivity(i)
    }

    fun openCamera(view: View) {
        val i = Intent()
        i.setAction(MediaStore.ACTION_IMAGE_CAPTURE)
        startActivityForResult(i, 90)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (resultCode == Activity.RESULT_OK && data != null) {
        }
    }
}

